### PR TITLE
Add clustering algorithm

### PR DESCRIPTION
**Pull request by GiorgioAlbertoLucia**
Branch with clustering algorithm added

### CODE REVIEW

**Review by Roberta-Ferioli — APPROVED** [review]

File: .vscode/settings.json
Comment (fchinu) [inline .vscode/settings.json:1]: Could you please add .vscode to .gitignore?

**Review by fchinu — CHANGES_REQUESTED** [review]
Thanks @GiorgioAlbertoLucia ! I will check the rest of the code by this evening, I promise

**Review by fchinu — CHANGES_REQUESTED** [review]
Thanks a lot @GiorgioAlbertoLucia! Last thing and I approve: can you remove .vscode from the repo?

**Comment by GiorgioAlbertoLucia** [comment]
Done

**Review by fchinu — APPROVED** [review]
Thanks a lot!

**Pull request merged**